### PR TITLE
[Bugfix] Issue 1041 - Displaying live charts over 4 hours fails

### PR DESCRIPTION
**Pull request by chess-equality**
https://github.com/sourceplusplus/sourceplusplus/issues/1041

There were overflows when `reservoirSize` = 720

image

This fixes them:

image

image

### CODE REVIEW

**Comment by BFergerson** [comment]
@chess-equality, is this a draft? The chart you show in the last picture is still very messed up.

**Comment by chess-equality** [comment]
@BFergerson I will check and fix https://github.com/sourceplusplus/sourceplusplus/issues/1071, this might be affecting the chart.

**Comment by chess-equality** [comment]
https://github.com/sourceplusplus/sourceplusplus/issues/1071 has also has been fixed, logs of testing from steps = 4 hours -> 12 hours -> 24 hours back to 1 hour are below:

image

image

image

image

File: plugin/src/main/kotlin/spp/jetbrains/sourcemarker/view/window/LiveViewChartWindow.kt [hunk 1]
```diff
@@ -99,19 +99,25 @@ class LiveViewChartWindow(
...
-        viewService.getHistoricalMetrics(
-            liveView.entityIds.toList(),
-            liveView.viewConfig.viewMetrics,
-            step, start, stop, labels
-        ).onSuccess {
-            for (i in 0 until it.data.size()) {
-                val stepBucket = step.bucketFormatter.format(start.plusSeconds((step.seconds * i).toLong())).toLong()
-                val metricData = it.data.getJsonObject(i)
-                addMetric(metricData.put("timeBucket", stepBucket))
+        viewService.clearLiveViews()
```
Comment (BFergerson) [inline plugin/src/main/kotlin/spp/jetbrains/sourcemarker/view/window/LiveViewChartWindow.kt]: This is a fairly destructive act. Why can't you remove the single live view causing issues?

File: plugin/src/main/kotlin/spp/jetbrains/sourcemarker/view/window/LiveViewChartWindow.kt [hunk 2]
```diff
@@ -267,6 +273,15 @@ class LiveViewChartWindow(
         }
     }
 
+    private fun timeSizeMultiplier(): Int {
+        return when (step) {
+            MetricStep.SECOND -> reservoirSize*60
```
Comment (BFergerson) [inline plugin/src/main/kotlin/spp/jetbrains/sourcemarker/view/window/LiveViewChartWindow.kt]: You've got build errors coming from detekt. Run `plugin-detekt`

File: plugin/src/main/kotlin/spp/jetbrains/sourcemarker/view/window/LiveViewChartWindow.kt (hunk 1 above)
Comment (chess-equality) [reply plugin/src/main/kotlin/spp/jetbrains/sourcemarker/view/window/LiveViewChartWindow.kt]: https://github.com/sourceplusplus/sourceplusplus/issues/1071 looks to be fixed after update from master. Attached is log of `demo-java` testing various times 5 mins --> 24 hours back and forth, no errors.

[idea_issue-1071-fixed_demo-java.log](https://github.com/sourceplusplus/interface-jetbrains/files/11827875/idea_issue-1071-fixed_demo-java.log)

Comment (BFergerson) [reply plugin/src/main/kotlin/spp/jetbrains/sourcemarker/view/window/LiveViewChartWindow.kt]: How is this resolved? You're still using `clearLiveViews()`.

Comment (chess-equality) [reply plugin/src/main/kotlin/spp/jetbrains/sourcemarker/view/window/LiveViewChartWindow.kt]: Sorry I thought I have already taken it out.

**Pull request merged**